### PR TITLE
Document `CMD`-object customization

### LINKED ISSUE #19286: Manual: "Running External Programs" does not explain how to change working directory
I was surprised that this wasn't documented in the manual. Would be useful to have a section about `Cmd` objects and how you can customize them with the `dir` kwarg to set the working directory, and more generally the manipulation of working environments with `setenv()` and `withenv()`.

All of these are explained [in the standard library](http://docs.julialang.org/en/latest/stdlib/base/#Base.setenv), but it would be convenient to have a description in the manual, with suitable cross-references to the existing standard library documentation.

A simple example
```jl
Cmd(`pwd`, dir="..")
```

### PR DESCRIPTION
**Pull request by ChristianKurz**
fix #19286

### CODE REVIEW

**Comment by ChristianKurz** [comment]
CI Failures are unrelated:
* **Circle**  i686 & x86_64: `did not find test files for the following tests: OldPkg/pkg`

**Comment by KristofferC** [comment]
You need to rebase on master for Circle CI.

File: doc/src/manual/running-external-programs.md [hunk 1]
```diff
@@ -365,3 +365,14 @@ stages have different latency so they use a different number of parallel workers
 saturated throughput.
 
 We strongly encourage you to try all these examples to see how they work.
+
+## `CMD`-Objects
```
[the remark this answers is not in the document]
Comment (ChristianKurz) [reply doc/src/manual/running-external-programs.md]: no reason, just a habbit. Will switch to lowercase and remove the hyphen while i'm at it.

File: doc/src/manual/running-external-programs.md [hunk 2]
```diff
@@ -365,3 +365,14 @@ stages have different latency so they use a different number of parallel workers
...
+## `Cmd` Objects
+Instead of with backticks, [`Cmd`](@ref) objects can also be generated manually. This has the advantage,
+that arguments like the working directory and environment variables can be customized.
+
+A simple example:
+```julia
+Cmd(`pwd`, dir="..")
+```
+
+Also see [`setenv`](@ref), [`withenv`](@ref) for further information.
```
Comment (KristofferC) [inline doc/src/manual/running-external-programs.md]: Is it possible to be a bit more specific than "information"?

Also, how is `withenv` relevant here?

**Comment by vtjnash** [comment]
Replaced by https://github.com/JuliaLang/julia/pull/40539. Sorry this sat for so long!

**Pull request closed without merging**